notebooks/cookbook/Enhanced sampling methods/Umbrella sampling MD.ipynb

```diff
@@ -0,0 +1,436 @@
+    "                                  collision_rate=1.0/picosecond)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create the simulation object"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "platform = Platform.getPlatformByName('OpenCL')\n",
+    "platform.setPropertyDefaultValue('Precision', 'mixed')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "simulation = Simulation(pdb.topology,system, integrator, platform)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "simulation.reporters.append(DCDReporter('./'+current_file+''+ '.dcd', restart_freq))\n",
+    "simulation.reporters.append(CheckpointReporter('./'+current_file+''+ '.chk', min(prd_steps, 10*restart_freq)))\n",
+    "simulation.reporters.append(StateDataReporter(open('./log.' + current_file+'', 'w'), log_freq, step=True, potentialEnergy=True, kineticEnergy=True, totalEnergy=True, temperature=True, volume=True, density=True, speed=True))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Set up a single simulation with one umbrella (in reality you'd do multiple of these)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## set force constant K for the biasing potential.\n",
+    "## the unit here is kJ*mol^{-1}*nm^{-2}, which is the default unit used in OpenMM\n",
+    "K = 100\n",
+    "simulation.context.setParameter(\"k\", K)\n",
+    "\n",
+    "## M centers of harmonic biasing potentials\n",
+    "M = 20\n",
+    "r0_range = np.linspace(0.3, 2.0, M, endpoint = False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "simulation.context.setParameter('r0_d1', r0_range[1])\n",
+    "simulation.context.setParameter('r0_d2', r0_range[2])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  initial : -115004.21780371421 kJ/mol\n",
+      "  final : -145473.62098196533 kJ/mol\n"
+     ]
+    }
+   ],
+   "source": [
+    "simulation.context.setPositions(pdb.positions)\n",
+    "print('  initial : %s' % (simulation.context.getState(getEnergy=True).getPotentialEnergy()))\n",
+    "simulation.minimizeEnergy()\n",
+    "print('  final : %s' % (simulation.context.getState(getEnergy=True).getPotentialEnergy()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for i in range(5):\n",
+    "    simulation.step(10)"
```

Comment:
Explain what's going on.  We've set up a simulation with restraints.  Now we run some dynamics and collect statistics.  What statistics do we need to collect?  How do we collect them?  What do we do with them once we've collected them?